app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -208,57 +240,43 @@ def validate(self, data):
                 raise serializers.ValidationError(
                     {
                         "experiment": [
-                            f"Nimbus Experiment has status '{status}', but can only "
-                            f"be changed when set to '{required_statuses}'."
+                            f"Nimbus Experiment has status {status}, but can only "
+                            f"be changed when set to {required_statuses}."
                         ]
                     }
                 )
 
         return data
 
-    def validate_status(self, value):
-        if (
-            self.instance
-            and value != self.instance.status
-            and value not in self.VALID_STATUS_TRANSITIONS.get(self.instance.status, ())
-        ):
-            status = self.instance.status
-            raise serializers.ValidationError(
-                f"Nimbus Experiment status cannot transition from {status} to {value}."
-            )
-
-        return value
 
+class NimbusStatusTransitionValidator:
+    requires_context = True
 
-class NimbusDocumentationLinkSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = NimbusDocumentationLink
-        fields = (
-            "title",
-            "link",
-        )
+    def __init__(self, field, transitions):
+        self.field = field
+        self.transitions = transitions
 
+    def __call__(self, value, serializer):
+        if serializer.instance and value:
```

Comment:
The third parameter here in a class-based per-field validator __call__ isn't the serializer - it's an individual serializer_field definition. Scratching my head at the docs, but I'm not sure you can access an existing value or determine whether the parent serializer has a instance from a per-field validator context.

My suggestion was to define this transition validation as a standalone function and use that within each mixin - e.g. something like def validate_status_transition(status_from, status_to, allowed_transitions)

Reply:
I think this makes sense. Thank you!

Reply:
Oh, but also I just poked around the DRF source a bit and found [this promising stretch of code](https://github.com/encode/django-rest-framework/blob/d635bc9c71b6cf84d137a68610ae2e628f8b62b3/rest_framework/validators.py#L67-L68):

```python
    def __call__(self, value, serializer_field):
        # Determine the underlying model field name. This may not be the
        # same as the serializer field name if `source=<>` is set.
        field_name = serializer_field.source_attrs[-1]
        # Determine the existing instance, if this is an update operation.
        instance = getattr(serializer_field.parent, 'instance', None)
```